snap/snapcraft.yaml

```diff
@@ -79,12 +79,45 @@ parts:
       # Replace key-value store path:
       sed -i 's/\/tmp/\/mnt/g' src/platform/Linux/CHIPPlatformConfig.h
 
+      # Install NodeJS and ZAP tool for arm builds
+      if [[ $SNAP_ARCH == "arm64" ]]; then
+        set -x
+
+        mkdir node_js
+        cd node_js
+        wget https://nodejs.org/dist/v12.22.12/node-v12.22.12-linux-x64.tar.xz
+        tar xfvJ node-v12.22.12-linux-x64.tar.xz
+        cp -rn node-v12.22.12-linux-x64/. /opt/node-v12.22.12-linux-x64/
+        rm -r node-v12.22.12-linux-x64
+        rm -rf /opt/node
+        rm -rf /usr/bin/node
+        rm -rf /usr/bin/npm
+        rm -rf /usr/bin/npx
+        ln -s /opt/node-v12.22.12-linux-x64 /opt/node
+        ln -s /opt/node/bin/* /usr/bin
+        cd ..
+        rm -rf node_js
+
+        ZAP_VERSION=v2023.05.22-nightly
+        mkdir -p /opt/zap-${ZAP_VERSION}
+        git clone https://github.com/project-chip/zap.git /opt/zap-${ZAP_VERSION}
+        cd /opt/zap-${ZAP_VERSION}
+        git checkout -b ${ZAP_VERSION}
+        npm cache clean --force
+        npm install -g npm@latest
+        npm ci
```

Comment:
This command can run successfully on amd64, but when attempting an arm64 snapcraft remote build, it consistently throws the following network-related error:
```
:: + npm ci
...
:: npm ERR! code E501
:: npm ERR! 501 Gateway error - GET https://registry.npmjs.org/yargs-parser/-/yargs-parser-18.1.3.tgz
```

Reply:
The error might be due to the network setup in Snapcraft remote build. I have successfully built it locally on Raspberry Pi 4B without any errors.

Reply:
Thanks for tracking down the issue.
For now, please release your local build to a branch.